contexts/SecurityProvider.tsx

```diff
@@ -0,0 +1,305 @@
+import crypto from 'crypto';
+import useKeychain from '../utils/Keychain';
+import EncryptedStorage from 'react-native-encrypted-storage';
+import LocalAuth from '../utils/LocalAuth';
+import {Text} from '@tsejerome/ui-kitten-components';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import Container from '../components/Container';
+import {scale} from 'react-native-size-matters';
+import useLockedScreen from '../hooks/useLockedScreen';
+import useAsyncStorage from '../hooks/useAsyncStorage';
+import useWallet from '../hooks/useWallet';
+import {AppState} from 'react-native';
+import {useNavigation} from '@react-navigation/native';
+
+export const SecurityProvider = (props: any) => {
+  const {lockedScreen, setLockedScreen} = useLockedScreen();
+  const [lockAfterBackground, setLockAfterBackground] = useAsyncStorage(
+    'lockAfterBackground',
+    'false',
+  );
+  const {navigate} = useNavigation();
+
+  const {refreshWallet} = useWallet();
+
+  const appState = useRef(AppState.currentState);
+
+  const [appStateVisible, setAppStateVisible] = useState(appState.current);
+
+  useEffect(() => {
+    if (appStateVisible == 'active' && lockedScreen) {
+      LocalAuth((error: any) => {
+        if (!error) {
+          setLockedScreen(false);
+        } else {
+          setLockedScreen(true);
+        }
+      });
+    }
+  }, [appStateVisible]);
+
+  useEffect(() => {
+    const subscription = AppState.addEventListener('change', nextAppState => {
+      DeviceInfo.isPinOrFingerprintSet().then(setIsPinOrFingerprintSet);
+      Keychain.getSupportedBiometryType({}).then(setSupportedBiometry);
+      if (appState.current.match(/background/) && nextAppState === 'active') {
+        if (refreshWallet) {
+          refreshWallet();
+        }
+        if (lockAfterBackground === 'true') {
+          setLockedScreen(true);
+        }
+      }
+      appState.current = nextAppState;
+      setAppStateVisible(appState.current);
+    });
+
+    return () => {
+      subscription.remove();
+    };
+  }, [refreshWallet, lockAfterBackground, setLockedScreen]);
+
+  const {read} = useKeychain();
+
+  const [isPinOrFingerprintSet, setIsPinOrFingerprintSet] = useState<
+    boolean | undefined
+  >(undefined);
+
+  const [supportedBiometry, setSupportedBiometry] =
+    useState<Keychain.BIOMETRY_TYPE | null>(null);
+
+  useEffect((): void => {
+    DeviceInfo.isPinOrFingerprintSet().then(setIsPinOrFingerprintSet);
+    Keychain.getSupportedBiometryType({}).then(setSupportedBiometry);
+  }, []);
+
+  const [setManualPin, setSetManualPin] = useState<any>(() => {});
+  const [askManualPin, setAskManualPin] = useState<any>(() => {});
+
+  const [supportedType, setSupportedType] =
+    useState<SecurityAuthenticationTypes>(SecurityAuthenticationTypes.NONE);
+
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  useEffect((): void => {
+    AsyncStorage.getItem('AuthenticationType').then(async val => {
```

Comment:
when are we using `AsyncStorage.getItem` and when are we using `useAsyncStorage`? we shd stick with one of them

Reply:
yep, I will move to use `useAsyncStorage`